src/main/java/address/events/LoadPrefsRequestEvent.java

```diff
@@ -0,0 +1,20 @@
+package address.events;
+
+import java.io.File;
+
+/**
+ * Indicates a request to load prefs from the file
+ */
+public class LoadPrefsRequestEvent extends BaseEvent {
+    /** The file from which the prefs to be loaded */
+    public File file;
+
+    public LoadPrefsRequestEvent(File file) {
+        this.file = file;
+    }
+
+    @Override
+    public String toString(){
+        return "from " + file.getAbsolutePath();
+    }
+}
```

Comment:
Seems that's impossible, though. Because `StorageManager` needs `ModelManager` to be passed to its constructor. I can still load the prefs from `StorageManager` by calling `ModelManager.updatePrefsFromExternalData`, if that's fine with you.

Otherwise, `ModelManager` will need to update addressbook using event. (currently `StorageManager` will call `ModelManager.updateUsingExternalData` upon handling load event).